### PR TITLE
Frontend validation for org and beneficiary creation

### PR DESCRIPTION
**Pull request by opishcheiko**
- form validation for org and beneficiary. Validation is very simple, if any of required fields is empty, it will show this error message
Screenshot 2021-03-05 at 14 11 27

- list of all orgs in admin
- list of collaborators and beneficiaries in admin

### CODE REVIEW

File: pages/admin/all-organizations.tsx [hunk 1]
```diff
@@ -23,57 +26,63 @@ import { DataGrid, ColDef, ValueGetterParams } from '@material-ui/data-grid'
...
+    },
+    {
+      field: 'button3',
+      headerName: '*',
+      renderCell: (params: ValueGetterParams) =>  (
+        <Button variant="outlined" href={`/${params.getValue('id')}/admin`}>Org admin</Button>
+       ),
+      sortable: false,
+      width: 150,
+    },
```
Comment (Jnegrier) [inline pages/admin/all-organizations.tsx:71]: This looks advanced, will have to study it later :)

File: pages/create-organization.tsx [hunk 2]
```diff
@@ -140,17 +139,71 @@ function Index() {
...
+    },
+    email: ""
   })
 
+  const [validationError, setValidationError] = React.useState(0);
+
+  function validateOrganizationDetails () {
+    if (organizationDetails.name === "" || 
+    organizationDetails.mission === "" || 
+    // organizationDetails.logo_url === "" ||  //TODO: figure out how to handle this validation
```
Comment (Jnegrier) [inline pages/create-organization.tsx]: Yeah, it should be just a string, but I noticed that sometimes the state is not set due some reason

**Review by Jnegrier — APPROVED** [review]
Looks amazing, thanks a million for working on the validation, it's really helpful, I will add some validation on the backend as well 👍

**Comment by Jnegrier** [comment]
Hi Sasha, I did two things in the last commit:

- A simple message to show if there are any errors coming from the backend, it will replace the happy message about org created, with a list of errors
- The issue about the last character not being updated on the `onChange` event is fixed :) it turns out that setting an state is an async operation, so when we were setting the full object back to the parent, it was still not fully updated. So the solution is to apply `useEffect` to update the object in the parent:

Something like this:

```
function SuperFunction(props: Props) {
  const { callback } = props

  const [orgData, setOrgData] = React.useState({
    <things>
  });

  function updateForm(type, data) {
    setOrgData({ ...orgData, [type]: data })
  }

  useEffect(()=>{
    callback(orgData);
  }, [orgData])

}

```

The only problem still missing in this form, is that when you upload a picture, the state with the rest of the data gets reset, and only the URL of the image keeps stored in the object, I tried to fix it, but didn't find an easy way

**Comment by opishcheiko** [comment]
@Jnegrier , ~i think the easy way is to put media into a separate step/separatee object~ Actually, I have figured out: I will create a separate step for now, but in theory, it should also work as it is. 

Thanks for fixing this last char cropping, team work 👊🏽

**Comment by Jnegrier** [comment]
> @Jnegrier , ~i think the easy way is to put media into a separate step/separatee object~ Actually, I have figured out: I will create a separate step for now, but in theory, it should also work as it is.
> 
> Thanks for fixing this last char cropping, team work 👊🏽

Yeah, there should be a way to make it work, I'm also happy to leave it as an independent step for the meantime, please feel free to merge whenever you see fit 👍

**Comment by opishcheiko** [comment]
> > @Jnegrier , ~i think the easy way is to put media into a separate step/separatee object~ Actually, I have figured out: I will create a separate step for now, but in theory, it should also work as it is.
> > Thanks for fixing this last char cropping, team work 👊🏽
> 
> Yeah, there should be a way to make it work, I'm also happy to leave it as an independent step for the meantime, please feel free to merge whenever you see fit 👍

I think I understood what's the issue. The object that's returned from PhotoId component is returned empty + url. Instead of just adding a url to existing data. So my solution would be:

when we update media url, to list all the attributes of the object, not just that url:
```
function updateLogoUrl(data) {
    setOrgMedia({
 ["logo_url"]: data, 
["photo_id_url"]: orgMedia.photo_id_url, 
["name"]: orgMedia.name, 
["orgId"]: orgMedia.orgId, 
["country"]: orgMedia.country})
  }
``` 
or somehow fix that url callback to return the full object

**Comment by Jnegrier** [comment]
> > > @Jnegrier , ~i think the easy way is to put media into a separate step/separatee object~ Actually, I have figured out: I will create a separate step for now, but in theory, it should also work as it is.
> > > Thanks for fixing this last char cropping, team work 👊🏽
> > 
> > Yeah, there should be a way to make it work, I'm also happy to leave it as an independent step for the meantime, please feel free to merge whenever you see fit 👍
> 
> I think I understood what's the issue. The object that's returned from PhotoId component is returned empty + url. Instead of just adding a url to existing data. So my solution would be:
> 
> when we update media url, to list all the attributes of the object, not just that url:
> ```
> function updateLogoUrl(data) {
>     setOrgMedia({
>  ["logo_url"]: data, 
> ["photo_id_url"]: orgMedia.photo_id_url, 
> ["name"]: orgMedia.name, 
> ["orgId"]: orgMedia.orgId, 
> ["country"]: orgMedia.country})
>   }
> ``` 
> or somehow fix that url callback to return the full object

Cool! Thanks for figuring it out!

**Pull request merged**